### PR TITLE
add rocmcc compiler based on aocc and clang

### PR DESCRIPTION
**Pull request by becker33**
Compiler definition for rocmcc based on the existing definitions for aocc and clang.

AMD has two separate products `aocc` and `rocm`, both providing compilers based on `clang`. Spack currently only has one compiler `aocc` with which to model them. This PR adds the needed `rocmcc` definition.

### CODE REVIEW

**Comment by lukebroskop** [comment]
@becker33  I may be missing something, but it looks like this PR does not add compiler wrappers to spack:
```
lukebr@crusher020:~ ia /gpfs/alpine/ven114/scratch/lukebr/confidence/spack/lib/spack/env/
aocc/             case-insensitive/ f77               ftn               ld.lld            xl/
arm/              cc                f90               gcc/              nag/              xl_r/
c++               cce/              f95               intel/            nvhpc/            
c89               clang/            fc                ld                oneapi/           
c99               cpp               fj/               ld.gold           pgi/    
```
shouldn't there be a rocmcc dir with compiler wrappers?

**Comment by becker33** [comment]
@lukebroskop good catch. Added

**Comment by becker33** [comment]
@spackbot fix style

File: lib/spack/spack/compilers/rocmcc.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    PrgEnv = 'PrgEnv-amd'
+    PrgEnv_compiler = 'amd'
+
+    @property
+    def link_paths(self):
+        link_paths = {'cc': 'rocmcc/clang',
+                      'cxx': 'rocmcc/clang++',
+                      'f77': 'rocmcc/flang',
+                      'fc': 'rocmcc/flang'}
```
Comment (lukebroskop) [inline lib/spack/spack/compilers/rocmcc.py]: @becker33 I think you want to prepend the wrappers with `amd`
e.g. 
```
        link_paths = {'cc': 'rocmcc/amdclang',
                      'cxx': 'rocmcc/amdclang++',
                      'f77': 'rocmcc/amdflang',
                      'fc': 'rocmcc/amdflang'}
```

Comment (lukebroskop) [reply lib/spack/spack/compilers/rocmcc.py]: @becker33  one more issue I encountered during testing concerns `lib/spack/env/cc`. I fixed the issue in #28579

Comment (trws) [inline lib/spack/spack/compilers/rocmcc.py]:
```suggestion
        link_paths = {'cc': 'rocmcc/amdclang',
                      'cxx': 'rocmcc/amdclang++',
                      'f77': 'rocmcc/amdflang',
                      'fc': 'rocmcc/amdflang'}
```

**Comment by trws** [comment]
What's still needed here? A lack of this is currently causing some pain in another build-related project.

**Review by tgamblin — APPROVED** [review]

**Comment by tgamblin** [comment]
I verified this works on corona.

**Pull request merged**